cadc-dali/src/main/java/ca/nrc/cadc/dali/util/DoubleArray2DFormat.java

```diff
@@ -0,0 +1,169 @@
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 4 $
+ *
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.dali.util;
+
+import java.util.Iterator;
+
+/**
+ * Formats and parses a double[].
+ *
+ */
+public class DoubleArray2DFormat implements Format<double[][]>
+{
+    private int[] arrayshape;
+    
+    public DoubleArray2DFormat(int[] arrayshape)
+    {
+        this.arrayshape = arrayshape;
+    }
+    
+    /**
+     * Takes an double[] and returns the standard String representation.
+     * If the double[] is null an empty String is returned.
+     *
+     * @param object double[] to format.
+     * @return String representation of the double[].
+     */
+    public String format(double[][] object)
+    {
+        if (object == null)
+        {
+            return "";
+        }
+        
+        StringBuilder sb = new StringBuilder();
+        for (int i=0; i<object.length; i++)
+            for (int j=0; j<object[i].length; j++)
+            {
+                sb.append(Double.toString(object[i][j]));
+                sb.append(" ");
+            }
+        return sb.substring(0, sb.length() - 1); // trim trailing comma
+    }
```

Comment:
trailing space hence a trim() would be more suitable maybe.

Reply:
I changed a bunch of array formatters to use StringBuilder substring to trimToSize and toString; the separator used to be comma hence the old trim hack :-)